conjure-core/src/test/java/com/palantir/conjure/parser/types/TypeParserTests.java

```diff
@@ -85,6 +85,9 @@ public void testParser_refType() throws ParseException {
     public void testParser_foreignRefType() throws ParseException {
         assertThat(TypeParser.INSTANCE.parse("bar.Foo"))
                 .isEqualTo(ForeignReferenceType.of(Namespace.of("bar"), TypeName.of("Foo")));
+
+        assertThat(TypeParser.INSTANCE.parse("bar_1.Foo"))
+                .isEqualTo(ForeignReferenceType.of(Namespace.of("bar_1"), TypeName.of("Foo")));
```

Comment:
Can we add a test that covers the `notAllowedToStartWith` codepath e.g. for `1_bar.Foo`?